backup-stores/s3/src/main/java/io/camunda/zeebe/backup/s3/util/AsyncAggregatingSubscriber.java

```diff
@@ -59,18 +60,25 @@ public void onNext(final CompletableFuture<T> future) {
           if (throwable == null) {
             LOG.trace("Completed: {}", result);
             results.add(result);
+            phaser.arrive();
+            if (!phaser.isTerminated()) {
+              subscription.request(1);
+            } else {
+              subscription.cancel();
```

Comment:
:x: When one of the futures fails, we can immediately cancel the subscription and don't need to wait for the next failing future to see a terminated phaser before cancelling the subscription. So we should move this to the else block below.